src/modules/paymentProcessor/PP_Queue_v1.sol

```diff
@@ -0,0 +1,941 @@
+                requiredDataLength++;
+            }
+        }
+
+        // Verify data array length matches number of set flags.
+        if (data_.length < requiredDataLength) {
+            revert Module__PP_Queue_InvalidFlagsOrData(flags_, data_.length);
+        }
+
+        // Validate each flag's data based on type.
+        uint dataIndex = 0;
+
+        // Check orderID flag (bit 0).
+        if (flagsValue & (1 << FLAG_ORDER_ID) != 0) {
+            // orderID must be non-zero.
+            if (uint(data_[dataIndex++]) == 0) {
+                revert Module__PP_Queue_InvalidFlagsOrData(flags_, data_.length);
+            }
+        }
+
+        // Check start time flag (bit 1).
+        if (flagsValue & (1 << FLAG_START_TIME) != 0) {
+            // start time must be in the future or 0.
+            uint startTime = uint(data_[dataIndex++]);
+            if (startTime != 0 && startTime < block.timestamp) {
+                revert Module__PP_Queue_InvalidFlagsOrData(flags_, data_.length);
+            }
+        }
+
+        // Check cliff period flag (bit 2).
+        if (flagsValue & (1 << FLAG_CLIFF_PERIOD) != 0) {
+            dataIndex++; // cliff period can be any value.
+        }
+
+        // Check end time flag (bit 3).
+        if (flagsValue & (1 << FLAG_END_TIME) != 0) {
+            // end time must be after start time if both are specified.
+            if (flagsValue & (1 << FLAG_START_TIME) != 0) {
+                uint endTime = uint(data_[dataIndex]);
+                // start time is second element if present.
+                uint startTime = uint(data_[1]);
+                if (endTime <= startTime) {
+                    revert Module__PP_Queue_InvalidFlagsOrData(
+                        flags_, data_.length
+                    );
+                }
+            }
+        }
+    }
+
+    /// @dev    Internal function to check whether the client is valid.
+    /// @param  client_ Address to validate.
+    function _ensureValidClient(address client_) internal view {
+        if (client_ == address(0)) {
+            revert Module__PP_Queue_InvalidClientAddress(client_);
+        }
+        if (client_ != _msgSender()) {
+            revert Module__PP_Queue_OnlyCallableByClient();
+        }
+    }
+
+    /// @dev    Sets the unclaimable amount for a specific payment.
+    /// @param  client_ The client address.
+    /// @param  token_ The token address.
+    /// @param  receiver_ The receiver address.
+    /// @param  amount_ The amount to set.
+    function _setUnclaimableAmount(
+        address client_,
+        address token_,
+        address receiver_,
+        uint amount_
+    ) internal {
+        _unclaimableAmountsForRecipient[client_][token_][receiver_] = amount_;
+    }
+
+    /// @dev    Checks if an order exists.
+    /// @param  orderId_ ID of the order to check.
+    /// @return exists_ True if the order exists.
+    function _orderExists(uint orderId_) internal view returns (bool exists_) {
```

Comment:
Would not work if we go ahead with the hashed order ID. So the checks in `getOrder` & `cancelPaymentOrderThoughQueueId` would need be removed/changed

Reply:
We move forward without hashing the ID, so this comment can be ignored